tests/PHPStan/Rules/Properties/TypesAssignedToPropertiesRuleTest.php

```diff
@@ -139,6 +139,16 @@ public function testBug1216(): void
 		]);
 	}
 
+	public function testBug1311(): void
+	{
+		$this->analyse([__DIR__ . '/data/bug-1311.php'], [
+			[
+				'Property Bug1311\HelloWorld::$list (array<int, string>) does not accept array<int, int>.',
```

Comment:
this is still here in a different format, because the test basically has strict rules enabled. not fully sure how to deal with this

Reply:
tbh I was hoping you would tell me 😅 looks like it's coming from the $strictTypes here: https://github.com/phpstan/phpstan-src/blob/c8833df6ffe3de490c641db572fc68a65bf412f2/src/Type/StringType.php#L128-L130

but not sure how that is related to strict-rules..

Reply:
aha on the playground the treatPhpDocTypesAsCertain is also somehow involved apparently. I obviously did not invest carefully now but it's confusing :)

Reply:
you were right, setting `shouldPolluteScopeWithLoopInitialAssignments` to `true` makes the error go away in the test